### DISCUSSION TITLE
When should `item.word` be used for a completion item?

### DISCUSSION THREAD

**Discussion by smjonas** [hrsh7th/nvim-cmp#657] [Q&A] [answered]
I understand what `label` and `insertText` are but when should `item.word` be used in a completion item? What does it do?
https://github.com/hrsh7th/nvim-cmp/blob/a58712bf16a61b2ef7260c2df7c732e3e2313392/lua/cmp/types/lsp.lua#L192-L197

**Comment by dmitmel** [comment]
(Not a definitive answer) So, the only places where this field is read are:

https://github.com/hrsh7th/nvim-cmp/blob/b11f8bbee3d7ba5190b043e23bd6f5b9cb82382c/lua/cmp/entry.lua#L105-L107

https://github.com/hrsh7th/nvim-cmp/blob/b11f8bbee3d7ba5190b043e23bd6f5b9cb82382c/lua/cmp/core.lua#L377

which leads me to believe that it is just a nvim-cmp-specific override of the `label` field (remember that the value of `label` is used as a fallback value if `insertText`, or `filterText` etc are not present, so the value of `word` will also propagate to those fields). As for when you should use it and not `label` - no idea. It has been there since the initial commit.

**Reply by smjonas** [reply]
Thank you, I also treated it the same as `label`. I saw that it is now used in cmp-path since the last commit and was not sure if it should be set for my source as well. Maybe @hrsh7th will have some additional clarification on this :)

**Comment by hrsh7th** [comment, marked as answer]
First, I would like to remove the word and dup properties if possible.

But at the moment they are implemented.
First, the `dup` property is needed because LSP doesn't have the ability to" remove the candidate if there is another high-priority identical word "in vim.

And the reason the `word` property exists is that LSP and VSCode don't define" text to insert at selection ".

_google translated_

**Comment by hrsh7th** [comment]
This implementation and hacky solutions should be removed, I think. But I don't have solution at the moment...

**Comment by hrsh7th** [comment]
|Purpose|LSP|Vim|nvim-cmp|
|-----|-----|-----|-----|
|Text for selection|N/A|word|Use word if provieded. if no, should calculate it by hacky implementation (like coc)|
|Filtering text|filterText or label|word or abbr|Use suitable property as defined in LSP|
|Text for confirmation|textEdit or insertText or label|N/A|Use suitable property as defined in LSP|
